SQRLUtilsLib/SQRL.cs

```diff
@@ -953,67 +1008,79 @@ public SQRLServerResponse GenerateQueryCommand(Uri sqrl, KeyPair siteKP, SQRLOpt
         }
 
         /// <summary>
-        /// Sends a new Identity (along with VUK/SUK) to server along with an ident command
+        /// Sends a new identity (along with VUK/SUK) to server along with an ident command
+        /// and returns the server's response.
         /// </summary>
-        /// <param name="sqrl"></param>
-        /// <param name="siteKP"></param>
-        /// <param name="encodedServerMessage"></param>
-        /// <param name="ilk"></param>
-        /// <param name="opts"></param>
-        /// <returns></returns>
+        /// <param name="sqrl">The SQRL server URI.</param>
+        /// <param name="siteKP">The site-specific ECDH public-private key pair.</param>
+        /// <param name="encodedServerMessage">The base64_url-encoded contents for the server parameter.</param>
+        /// <param name="ilk">The Identity Lock Key (ILK).</param>
+        /// <param name="opts">Optional SQRL options to be appended to the client message.</param>
+        /// <param name="sin">Optional base64_url-encoded Secret Index (SIN) parameter to be appended to the client message.</param>
+        /// <returns>Returns a <c>SQRLServerResponse</c> object representing the server's response details.</returns>
         public SQRLServerResponse GenerateNewIdentCommand(Uri sqrl, KeyPair siteKP, string encodedServerMessage, byte[] ilk, SQRLOptions opts = null, StringBuilder sin=null)
         {
-
-            SQRLServerResponse serverResponse = null;
             var sukvuk = GetSukVuk(ilk);
             StringBuilder addClientData = new StringBuilder();
             addClientData.AppendLineWindows($"suk={Sodium.Utilities.BinaryToBase64(sukvuk.Key, Sodium.Utilities.Base64Variant.UrlSafeNoPadding)}");
             addClientData.AppendLineWindows($"vuk={Sodium.Utilities.BinaryToBase64(sukvuk.Value, Sodium.Utilities.Base64Variant.UrlSafeNoPadding)}");
             if (sin != null)
                 addClientData.Append(sin);
-            serverResponse = GenerateSQRLCommand(SQRLCommands.ident, sqrl, siteKP, encodedServerMessage, addClientData, opts, null);
 
-            return serverResponse;
+            return GenerateSQRLCommand(SQRLCommands.ident, sqrl, siteKP, encodedServerMessage, addClientData, opts, null);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="sqrl">The SQRL server URI.</param>
+        /// <param name="siteKP">The site-specific ECDH public-private key pair.</param>
+        /// <param name="encodedServerMessage">The base64_url-encoded contents for the server parameter.</param>
+        /// <param name="ilk">The Identity Lock Key (ILK).</param>
+        /// <param name="ursKey">The Unlock Request Signing Key (URSK).</param>
+        /// <param name="priorKey">The Previous Identity Key (PIDK) key pair to be appended to the client message.</param>
+        /// <param name="opts">Optional SQRL options to be appended to the client message.</param>
```

Comment:
This one does an Ident but signs it with the old URS and sends new SUK and VUK effectively replacing the old identity with a new rekeyed one

Reply:
Sorry, @josegomez, I saw your approval and thought you want me to merge the PR myself.
I'll still add that one to a later PR.

Reply:
No worries, I saw this little thing after I looked at it again, no bigdeal

Reply:
and yes fee free to merge anything as long as it passes the tests etc I have no issues with that